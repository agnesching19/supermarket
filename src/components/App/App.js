import React from 'react';
import Basket from '../Basket';
import Product from '../Product';
import Styled from './styled';

const App = () => (
  <Styled>
    <Product
      id='item_beans'
      name="Beans"
      url="https://mysupermarket.s3-eu-west-1.amazonaws.com/beans.png"
      promotion="Get 3 for 2!"
      unitPrice={0.50}
      priceByItem
    />
    <Product
      id='item_coke'
      name="Coke"
      url="https://mysupermarket.s3-eu-west-1.amazonaws.com/coke.jpg"
      promotion="Get 2 for 1!"
      unitPrice={0.70}
      priceByItem
    />
    <Product
      id='item_orange'
      name="Oranges"
      url="https://mysupermarket.s3-eu-west-1.amazonaws.com/oranges.jpg"
      unitPrice={1.99}
      priceByWeight
      weight={0}
    />
    <Basket />
  </Styled>
);

export default App;
