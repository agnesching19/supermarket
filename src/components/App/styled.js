import styled from 'styled-components';

export default styled.div`
  display: flex;
  margin: 0 auto;
  width: 960px;
`;