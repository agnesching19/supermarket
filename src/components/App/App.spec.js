import React from 'react';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import {render} from '@testing-library/react';
import App from './App';

describe('components > App', () => {
  const mockStore = configureStore([]);
  const store = mockStore({
    checkout: {
      showCheckout: false,
      total: {
        discount: null,
        totalBeforeDiscount: 0,
        totalFinal: 0,
      },
      basketItems: [
        {
          id: 'id-test',
          name: 'test',
        },
      ],
    },
  });

  it('renders without crashing', () => {
    const {asFragment} = render(<App />, {
      wrapper: ({children}) => <Provider store={store}>{children}</Provider>,
    });

    /**
     * Basic snapshot test to make sure, that rendered component
     * matches expected footprint.
     */
    expect(asFragment()).toMatchSnapshot();
  });
});
