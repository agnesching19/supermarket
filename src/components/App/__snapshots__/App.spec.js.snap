// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components > App renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="sc-AxiKw kYfILu"
  >
    <div
      class="sc-AxirZ hKwhGi"
    >
      <h4>
        Beans
      </h4>
      <img
        alt="Beans"
        src="https://mysupermarket.s3-eu-west-1.amazonaws.com/beans.png"
      />
      <div
        class="middle-div"
      >
        <p
          class="discount"
        >
          Get 3 for 2!
        </p>
      </div>
      <p
        class="unitPrice"
      >
        £0.50/each
      </p>
      <div
        class="bottom-div"
      >
        <button
          type="button"
        >
          +
        </button>
        <button
          type="button"
        >
          -
        </button>
      </div>
    </div>
    <div
      class="sc-AxirZ hKwhGi"
    >
      <h4>
        Coke
      </h4>
      <img
        alt="Coke"
        src="https://mysupermarket.s3-eu-west-1.amazonaws.com/coke.jpg"
      />
      <div
        class="middle-div"
      >
        <p
          class="discount"
        >
          Get 2 for 1!
        </p>
      </div>
      <p
        class="unitPrice"
      >
        £0.70/each
      </p>
      <div
        class="bottom-div"
      >
        <button
          type="button"
        >
          +
        </button>
        <button
          type="button"
        >
          -
        </button>
      </div>
    </div>
    <div
      class="sc-AxirZ hKwhGi"
    >
      <h4>
        Oranges
      </h4>
      <img
        alt="Oranges"
        src="https://mysupermarket.s3-eu-west-1.amazonaws.com/oranges.jpg"
      />
      <div
        class="middle-div"
      />
      <p
        class="unitPrice"
      >
        £1.99/kg
      </p>
      <div
        class="bottom-div"
      >
        <button
          type="button"
        >
          +
        </button>
        <button
          type="button"
        >
          -
        </button>
      </div>
    </div>
    <div
      class="sc-AxjAm cdEQCG"
    >
      <h2>
        Basket
      </h2>
      <ul
        class="list"
      >
        <li>
          <span
            class="item-total"
          >
            <span>
              test
            </span>
            <span
              class="item-price"
            >
              £0.00
            </span>
          </span>
        </li>
      </ul>
      <hr />
      <p
        class="sub-total"
        data-testid="sub-total"
      >
        <span>
          Sub total
        </span>
        <span>
          £0.00
        </span>
      </p>
      <button
        class="btn-checkout"
        data-testid="checkout"
        type="button"
      >
        Checkout
      </button>
      <ul
        class="total"
      />
    </div>
  </div>
</DocumentFragment>
`;
